examples/command.rs

```diff
@@ -0,0 +1,308 @@
+
+const SPAWN_Y: i32 = 64;
+
+pub enum TeleportResult {
+    ExecutorToLocation(arg_parser::Vec3),
+    ExecutorToTarget(String),
+    TargetToTarget((String, String)),
+    TargetToLocation((String, arg_parser::Vec3)),
+}
+
+#[derive(Resource, Clone)]
+struct TeleportCommand;
+
+impl Command for TeleportCommand {
+    type CommandExecutables = TeleportResult;
+
+    fn name() -> String {
+        "teleport".into()
+    }
+
+    fn assemble_graph(&self, command_graph: &mut CommandGraphBuilder<Self::CommandExecutables>) {
+        let teleport = command_graph
+            .root()
+            .literal("teleport")
+            .with_scopes(vec!["valence:command:teleport"])
+            .id();
+
+        // tp alias
+        command_graph
+            .root()
+            .literal("tp")
+            .with_scopes(vec!["valence:command:teleport"])
+            .redirect_to(teleport);
+
+        // executor to vec3 target
+        command_graph
+            .at(teleport)
+            .argument("destination:location")
+            .with_parser(Parser::Vec3)
+            .with_executable(|s| {
+                TeleportResult::ExecutorToLocation(arg_parser::Vec3::parse_args(s).unwrap())
+            });
+
+        // executor to entity target
+        command_graph
+            .at(teleport)
+            .argument("destination:entity")
+            .with_parser(Parser::Entity {
+                only_players: false,
+                single: true,
+            })
+            .with_executable(|s| {
+                TeleportResult::ExecutorToTarget(arg_parser::EntitySelector::parse_args(s).unwrap())
+            });
+
+        let targeted_teleport = command_graph
+            .root()
+            .at(teleport)
+            .argument("target:entity")
+            .with_parser(Parser::Entity {
+                only_players: false,
+                single: false,
+            })
+            .id();
+
+        // target to target
+        command_graph
+            .at(targeted_teleport)
+            .argument("destination:entity")
+            .with_parser(Parser::Entity {
+                only_players: false,
+                single: true,
+            })
+            .with_executable(|s| {
+                TeleportResult::TargetToTarget(
+                    <(arg_parser::EntitySelector, arg_parser::EntitySelector)>::parse_args(s)
+                        .unwrap(),
+                )
+            });
+        // target to location
+        command_graph
+            .at(targeted_teleport)
+            .argument("destination:location")
+            .with_parser(Parser::Vec3)
+            .with_executable(|s| {
+                TeleportResult::TargetToLocation(
+                    <(arg_parser::EntitySelector, arg_parser::Vec3)>::parse_args(s).unwrap(),
+                )
+            });
+    }
+}
+
+pub fn main() {
+    App::new()
+        .add_plugins((
+            DefaultPlugins,
+            CommandHandler::from_command(TeleportCommand),
```

Comment:
Don't think each command should be a separate plugin. This would clutter the list of loaded plugins pretty quickly.

What I'd like to see is an `add_command` method on `App` similar to `add_event`. You could do that via an `AddCommand` trait impl'd on `App` and maybe `World` too.

Reply:
What would the ad command function do aside from adding systems and/or plugins.